tests/integration/script.sh

```diff
@@ -0,0 +1,96 @@
+#!/bin/bash
+set -euo pipefail
+set +x
+
+# This test requires users to set up an azure service principle that has access
+# to an azure keyvault.
+# After setting AZURE_TENANT_ID, AZURE_CLIENT_ID, AZURE_CLIENT_SECRET environment
+# variables, users can run this script by providing the keyvault name like so:
+# ./tests/integration/script.sh <key-vault-name>
+# This script assumes users are running it from the root directory of the azure-plugin
+# repo
+cleanup(){
+    ret=$?
+    echo "EXIT STATUS: $ret"
+    git worktree prune
+    rm -rf "$TMP"
+    echo "cleaned up test successfully"
+    exit "$ret"
+}
+trap cleanup EXIT
+
+replace_in_file(){
+    local src="$1"
+    local dest="$2"
+    local path=$3
+    case "$OSTYPE" in
+    darwin*)
+        sed -i '' "s#${src}#${dest}#g" "${path}" ;;
+    *)
+        sed -i "s#${src}#${dest}#g" "${path}" ;;
+    esac
+}
+
+TMP=$(mktemp -d -t tmp.XXXXXXXXXX)
+
+vault_name=$1
+authSetup=0
+if [ -z ${AZURE_TENANT_ID} ]; then
+    echo "AZURE_TENANT_ID is required for authentication."
+	authSetup=1
+fi
+
+if [ -z ${AZURE_CLIENT_ID} ]; then
+    echo "AZURE_CLIENT_ID is required for authentication."
+	authSetup=1
+fi
+
+if [ -z ${AZURE_CLIENT_SECRET} ]; then
+    echo "AZURE_CLIENT_SECRET is required for authentication."
+	authSetup=1
+fi
+
+if [ $authSetup -eq 1 ]; then
+	exit 1
+fi
+
+if [ -z $vault_name ]; then
+    echo "argument vault name is required for running this test."
+	exit 1
+fi
+
+PORTER_HOME=${TMP}/bin/
+git fetch --no-tags --progress -- https://github.com/getporter/porter.git +refs/heads/release/v1:refs/remotes/origin/release/v1
+git worktree add -f "$PORTER_HOME" "release/v1"
```

Comment:
I received the following error when I ran the script:

```
$ ./tests/integration/script.sh carolynvs
remote: Enumerating objects: 28196, done.
remote: Counting objects: 100% (833/833), done.
remote: Compressing objects: 100% (612/612), done.
remote: Total 28196 (delta 375), reused 476 (delta 209), pack-reused 27363
Receiving objects: 100% (28196/28196), 34.61 MiB | 60.68 MiB/s, done.
Resolving deltas: 100% (16262/16262), done.
From https://github.com/getporter/porter
 * [new branch]        release/v1 -> origin/release/v1
fatal: invalid reference: release/v1
EXIT STATUS: 128
cleaned up test successfully
```

Instead of implementing here in a script installing porter, you can use the https://github.com/getporter/magefiles target `porter.EnsurePorter`. How about you add a target called `TestIntegration` and have it depend on EnsurePorter. Then it can call the bash script here.

Reply:
From my understanding, the `EnsurePorter` only works for published release versions? In this case, the test would fail since the new secret protocol is not in a release yet. 
What's your git version?

Reply:
I'm on 2.32

Reply:
I changed the reference to use the remote branch. Let me know if it still causes trouble for u

Reply:
The script is expecting me to set PORTER_HOME and when I set that to `$PWD/bin` it complains that it's not a working tree. Is this supposed to be the path to the porter repository or something that is set internally in the script and not by me?

Reply:
That's odd. What error do you get that prompt you to set PORTER_HOME manually? It should do that automatically by the script

Reply:
I finally got it to work by setting `export PORTER_HOME='. Maybe just remove the `set -u` flag, or move up the declaration of PORTER_HOME above cleanup?